### PR TITLE
updating trivy latest version and adding logging

### PR DESCRIPTION
**Pull request by rajithkrishnegowda**
This update includes:

Trivy Version Update: Upgraded Trivy to the latest version i.e. 0.24.0 to ensure that the latest security vulnerabilities are scanned effectively. This update helps maintain security compliance by using up-to-date vulnerability databases.

Enhanced Logging: Introduced comprehensive logging to improve visibility and debugging. The logging includes detailed scan results, error messages, and operational logs during the security scan. This ensures better monitoring and easier troubleshooting when analyzing scan outcomes.

### CODE REVIEW

**Review by rahulga1 — APPROVED** [review]

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -30,3 +30,17 @@ jobs:
...
+          image-ref: 'docker.io/securefederatedai/openfl:${{ github.sha }}'
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW'
+  
+      - name: Log completion message
+        run: |
+          echo "Trivy scan completed. Review the output above for vulnerabilities."
```
Comment (teoparvanov) [inline .github/workflows/trivy.yml:46]: Where are the trivy scan results published?

**Review by teoparvanov — APPROVED** [review]
LGTM, thanks @rajithkrishnegowda !

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (rajithkrishnegowda) [reply .github/workflows/trivy.yml:46]: in the above step it just echo the statements i.e echo "Trivy scan completed. Review the output above for vulnerabilities."

above this step we will Upload Trivy scan results to GitHub Security tab and also logs in the pipeline

Screenshot 2024-09-25 at 5 19 43 PM

**Review by ishaileshpant — APPROVED** [review]
LGTM

**Pull request merged**